packages/react-components/src/components/Accordion/hooks.ts

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+
+interface UseAccordionProps {
+  isControlled: boolean;
+  isExpanded: boolean;
+  setShouldBeVisible: React.Dispatch<React.SetStateAction<boolean>>;
```

Comment:
You can make it a simple callback with a parameter, so you do not need a react dispatch types here.